src/instrumentsscene/view/instrumentspaneltreemodel.cpp

```diff
@@ -414,6 +411,31 @@ bool InstrumentsPanelTreeModel::moveRows(const QModelIndex& sourceParent, int so
     return true;
 }
 
+void InstrumentsPanelTreeModel::changeVisibilityOfSelectedRows()
+{
+    if (!m_selectionModel || !m_selectionModel->hasSelection()) {
+        return;
+    }
+
+    QModelIndexList selectedIndexes = m_selectionModel->selectedIndexes();
+    int hiddenItems = 0, visibleItems = 0;
+    bool toVisible = true;
+
+    for (QModelIndex index : selectedIndexes) {
+        AbstractInstrumentsPanelTreeItem* item = modelIndexToItem(index);
+        item->isVisible() ? visibleItems++ : hiddenItems++;
+    }
+
+    if (visibleItems > hiddenItems) {
+        toVisible = false;
+    }
```

Comment:
Would it be an idea to base the new visibility on the old visibility of the instrument that was clicked, instead of on the number of visible selected instruments?

i.e. If there are two instruments selected, and the first is visible, but the second not, and you click on the eye icon for the first instrument, then both instruments become invisible, because the first instrument was visible.
And if you'd click the eye icon for the second instrument, both instruments become visible, because the second instrument was not visible. 

I think that would feel more predictable / easier for the user to figure out what the pattern is.

Reply:
Okay that seems like a better idea